internal/p2p/options_connection_logger.go

```diff
@@ -57,8 +58,9 @@ func (n *connectionLoggerNotifee) Connected(_ network.Network, conn network.Conn
 func (n *connectionLoggerNotifee) Disconnected(_ network.Network, conn network.Conn) {
 	n.log.
 		WithFields(log.Fields{
-			"peerID": conn.RemotePeer().String(),
-			"addr":   conn.RemoteMultiaddr().String(),
+			"peerID":    conn.RemotePeer().String(),
+			"addr":      conn.RemoteMultiaddr().String(),
+			"peerCount": len(n.ps.Peers()),
```

Comment:
Doesn't this duplicate the role of the `Monitor()`?

Reply:
not exactly
monitor gets the data every minute, but it sometimes misses when there is a short connection.
I want to have a heartbeat with the `Monitor()` and also log every event where someone connected or disconnected.

Reply:
![image](https://user-images.githubusercontent.com/6050167/125585902-94ed10bf-f9a5-49cf-b4c2-dc93e9953b50.png)
like this

Reply:
it is funny how we got 7.33

Reply:
So if this one is more precise, why do we need a `Monitor`? I still think this PR introduces two solutions to exactly the same problem. If you need `Monitor` to be more precise, then update `Monitor` to print logs more often.

Reply:
Yes, because it acts only on a connection event and they are fired rarely.
Monitor keeps the grafana charts alive and this part is responsible for logging at the precise moment of the occurence.

Reply:
What about the second part of my question?

Reply:
yes